components/server/src/workspace/headless-log-service.ts

```diff
@@ -311,6 +317,39 @@ export class HeadlessLogService {
                     if (!receivedDataYet && status.code === grpc.status.UNAVAILABLE) {
                         log.debug("stream headless workspace log", err);
                         reject(err);
+                        return;
```

Comment:
Conceptually the `server` changes make sense. Let's look in more detail once we aligned on where to put the supervisor API (see other comment). :+1: